pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -300,19 +297,19 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 	return config, nil
 }
 
-func buildKubeProxyConfig(options configapi.NodeConfig) (*proxyoptions.ProxyServerConfig, error) {
+func buildKubeProxyConfig(options configapi.NodeConfig, originClient *osclient.Client, kubeClient *client.Client) (*proxyoptions.ProxyServerConfig, osdnapi.FilteringEndpointsConfigHandler, error) {
```

Comment:
the filter isn't part of the proxy *config*... you could leave this as is and just add the call to `factory.NewProxyPlugin()` to `BuildKubernetesNodeConfig()`, right after the call to `factory.NewNodePlugin()`, and thereby simplify things a bunch...